pkg/server/router/tcp/router.go

```diff
@@ -61,10 +78,50 @@ func (m *Manager) BuildHandlers(rootCtx context.Context, entryPoints []string) m
 	return entryPointHandlers
 }
 
-func (m *Manager) buildEntryPointHandler(ctx context.Context, configs map[string]*config.TCPRouterInfo, handlerHTTP http.Handler, handlerHTTPS http.Handler) (*tcp.Router, error) {
+func (m *Manager) buildEntryPointHandler(ctx context.Context, configs map[string]*config.TCPRouterInfo, configsHTTP map[string]*config.RouterInfo, handlerHTTP http.Handler, handlerHTTPS http.Handler) (*tcp.Router, error) {
 	router := &tcp.Router{}
 	router.HTTPHandler(handlerHTTP)
-	router.HTTPSHandler(handlerHTTPS, m.tlsConfig)
+
+	defaultTLSConf, err := m.tlsManager.Get("default", "default")
+	if err != nil {
+		return nil, err
+	}
+
+	router.HTTPSHandler(handlerHTTPS, defaultTLSConf)
+
+	for routerHTTPName, routerHTTPConfig := range configsHTTP {
+		if len(routerHTTPConfig.TLS.Options) == 0 || routerHTTPConfig.TLS.Options == "default" {
+			continue
+		}
+
+		ctxRouter := log.With(internal.AddProviderInContext(ctx, routerHTTPName), log.Str(log.RouterName, routerHTTPName))
+		logger := log.FromContext(ctxRouter)
+
+		domains, err := rules.ParseDomains(routerHTTPConfig.Rule)
+		if err != nil {
+			routerErr := fmt.Errorf("unknown rule %s", routerHTTPConfig.Rule)
+			routerHTTPConfig.Err = routerErr.Error()
+			logger.Debug(routerErr)
+			continue
+		}
+
+		if len(domains) == 0 {
+			logger.Warnf("The 'default' tls option will be applied instead of %q as no domain has been defined", routerHTTPConfig.TLS.Options)
```

Comment:
```suggestion
			logger.Warnf("The 'default' tls options will be applied instead of %q as no domain has been found in the rule", routerHTTPConfig.TLS.Options)
```